test/test_st.t

```diff
@@ -382,8 +435,9 @@ Tests Model Tree:
   Execution mode: st
   
   >Program Print
-  ((AND (< (Val Int 3) (+ (Val Int 3) (Val Int 1))) (>= (Val Int 3) (Val Int 3))) (Val Int 5) (Val Int 0))
+  (bool.ite (bool.and true true) 5 0)
```

Comment:
This should be: 

(bool.ite (bool.and true true) 5 0) = 5

Missing simplifications in encoding